This patch fixes compiling xinetd without RPC support.

The content of this patch was copied from the OpenWrt project:
https://dev.openwrt.org/browser/packages/net/xinetd/patches/003-rpc_fix.patch

Signed-off-by: Danomi Manchego <danomimanchego123@gmail.com>

--- a/xinetd/confparse.c
+++ b/xinetd/confparse.c
@@ -745,7 +745,7 @@ static status_e check_entry( struct serv
 	   }
    }
 
-/* #ifndef NO_RPC */
+#ifndef NO_RPC
 #if defined(HAVE_RPC_RPCENT_H) || defined(HAVE_NETDB_H)
    if ( SC_IS_RPC( scp ) && !SC_IS_UNLISTED( scp ) )
    {
@@ -759,6 +759,7 @@ static status_e check_entry( struct serv
       SC_RPCDATA( scp )->rd_program_number = rep->r_number ;
    }
    else
+#endif
 #endif   /* ! NO_RPC */
    {
        if ( !SC_IS_UNLISTED( scp ) ) 
